lib/event/event_collection.rb

```diff
@@ -84,6 +84,18 @@ def as_json
     events.map(&:as_json)
   end
 
+  def self.from_array(events_as_array)
```

Comment:
As a class method, this belongs right at the top of the class, above `attr_reader`.